src/formulations/inverse-dynamics-formulation-acc-force.cpp

```diff
@@ -270,7 +270,7 @@ bool InverseDynamicsFormulationAccForce::updateRigidContactWeights(const std::st
     ConstraintLevel::iterator itt;
     bool force_reg_task_found = false;
     bool motion_task_found = false;
-    for(unsigned int i=1; i<m_hqpData.size(); i++)
+    for(unsigned int i=0; i<m_hqpData.size(); i++)
```

Comment:
I think the reason for skipping tasks at level zero is that they are just constraints, so changing the weights does not have any effect on them. Would you agree?

Reply:
I see, but by default the "motion_task" of the contact is at level zero and this test always fail: https://github.com/stack-of-tasks/tsid/blob/devel/src/formulations/inverse-dynamics-formulation-acc-force.cpp#L285

Reply:
I propose to revert this change of index but invert the 
`if(motion_weight>=0.0)` and `if(itt->second->name() == contact_name+"_motion_task")`.
That way it return False only if the user requested a change in the `motion_weight` while `motion_task` was in the level 0.

And the method can return True as expected if only a change of `force_regularization_weight` is requested.

Reply:
Yes, if the motion task is at level zero then its weight is not gonna be changed because changing it would not have any effect on the problem solution. However, one could also decide to have motion tasks at level 1, and in that case the weight would be changed. I don't see any problem with this, but maybe I am missing something.

Reply:
The problem is that in this case the method `updateRigidContactWeights` return False, which should the indication of a failure to apply the changes (or that something went wrong) while I was successfully executed.

Reply:
I reverted the changes of index and just changed the tests to return True in this case (`motion_task` in level 0 and no change of `motion_weight` requested).
Tell me if this change is OK for you.